scripts/deploy.sh

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env bash
+
+################################################################################
+# Following assumptions are made by this script:                               #
+# * gcloud, docker, and docker-compose is installed                            #
+# * it is called from $GOPATH/src/github.com/google/keytransparency/scripts    #
+# * there is a project called key-transparency on gce which has a cluster      #
+#   called "ci-cluster" within the "us-central1-a" compute zone                #
+# * there is a service key to authenticate with glcoud to above project in     #
+#   $GOPATH/src/github.com/google/keytransparency/service_key.json             #
+################################################################################
+
+PROJECT_NAME=key-transparency
+NAME_SPACE=default
+
+MAX_RETRY=30
+
+function main()
+{
+  # create key-pairs:
+  ./prepare_server.sh -f
+  initGcloud
+  buildDockerImgs
+  tearDown
+  pushTrillianImgs
+
+  # Deploy all trillian related services:
+  kubectl apply -f ../kubernetes/trillian-deployment.yml
+
+  pushKTImgs
+  waitForTrillian
+  createTreeAndSetIDs
+
+  # Deploy all keytransparency related services (server and signer):
+  kubectl apply -f ../kubernetes/keytransparency-deployment.yml
+}
+
+function initGcloud()
+{
+  gcloud --quiet version
+  gcloud auth activate-service-account --key-file ../service_key.json
+  # This might fail locally but is necessary on travis:
+  gcloud --quiet components update kubectl
+  gcloud config set project ${PROJECT_NAME}
+  gcloud config set compute/zone us-central1-a
+  gcloud container clusters get-credentials ci-cluster
+}
+
+function buildDockerImgs()
+{
+  # Build all images defined in the docker-compose.yml:
+  (cd ../ && docker-compose build)
+  # Separately build the DB:
+  # Work around some git permission issues on linux:
+  chmod a+r ../../trillian/storage/mysql/storage.sql
+  # Create and push a fresh db image (with schema created)
+  docker build -t  us.gcr.io/${PROJECT_NAME}/db -f ../kubernetes/mysql-trillian/Dockerfile ../..
```

Comment:
All done besides publishing. `gcloud` doesn't seem to support `docker-compose`. What we could do though, is make docker login into gcloud: https://stackoverflow.com/a/29311300 
Do you think that's safe and worth trying?